storagemarket/client.go

```diff
@@ -38,7 +38,7 @@ type StorageClient interface {
 	GetAsk(ctx context.Context, info StorageProviderInfo) (*SignedStorageAsk, error)
 
 	// GetProviderDealState queries a provider for the current state of a client's deal
-	GetProviderDealState(ctx context.Context, info StorageProviderInfo, proposalCid cid.Cid) (*ProviderDealState, error)
+	GetProviderDealState(ctx context.Context, deal ClientDeal) (*ProviderDealState, error)
```

Comment:
I feel pretty strongly this api should be:
```
GetProviderDealState(ctx context.Context, proposalCid cid.Cid) (*ProviderDealState, error)
```
you can make a private method on Provider with the current signature so you can use it for the state machine.
and then the implementation of GetProviderDealState should be pretty trivial:
```golang
func (c *Client) GetProviderDealState(ctx context.Context, proposalCid cid.Cid) (*storagemarket.ProviderDealState, error) {
var deal ClientDeal
err := c.statemachines.Get(proposalCID).Get(&deal)
if err != nil {
return nil, err
}
return c.getProviderDealState(ctx, deal)
}